### PR TITLE
Resend confirmation email

### PR DESCRIPTION
**Pull request by Timo614**
### References Issues:
References: #1390

### Description:
This branch deals with the associated issue by adding a new route to resend the order confirmation. I have the logic confirm the order is paid prior to sending and it uses the requires_scope_for_order logic to determine access via a new scope I added `OrderResendConfirmation`.

Changes:
- Moved the confirmation email mailer out of 'carts.rs' in favor of 'orders.rs' since it's no longer a cart at that point
- Fixed a bug with Orders#show controller that treated box office users as the owner of the order if viewing their own versus the on_behalf_of_user
- Added Orders#resend_confirmation, its route, and tests
- Fixed display_order's logic which tries to determine the remaining limit for a user (was using the user_id which in the case of the box office user would be their own id thus giving incorrect limits)
- Added a scope OrderResendConfirmation to more finely control access to the feature
- Fixed a display bug (see https://github.com/big-neon/bn-api/issues/1390#issuecomment-513841329) where we included the discount value in the fees totaling 
- Added support for having the resend confirmation show the latest state of the order after refunds
- Passing a new sendgrid variable `total_breakdown` that's unused at the moment to pass back the fee and total breakdown we currently breakout individually as sendgrid variables. This allows me to include the refund data (should likely be modified on the production template after we go live but the total listed now is refund inclusive so can wait a bit):

```
"total_breakdown":"<table><tbody><tr><th>Fees Total</th><td>1.50</td></tr><tr style=\"color: red\"><th>Refunded</th><td>($0.20)</td></tr><tr><th>Order Total</th><td>90.30</td></tr></tbody></table>",
```

Note:
- I didn't modify the order confirmation email at all in moving it. Waiting to confirm those questions prior to any modifications.

#### Examples

```
-- Draft order
curl --header "Content-Type: application/json" --request POST http://localhost:9000/orders/ff99422b-f837-4ed6-9dde-74b5f8bc7a7a/resend_confirmation --header "Authorization: Bearer eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJzdWIiOiJlNWMyOTRhMi00N2JjLTQyMjktYTUyNS0xMzFmYjY1MzZhNmIiLCJpc3MiOiJiZy1zYW1wbGUtaXNzdWVyIiwiZXhwIjoxNTYzNDcyNjU1fQ.Zf5wv_rVzMr-75kqXezjitCQrcZ56viStEz_vg8kfNs" -i
HTTP/1.1 500 Internal Server Error
content-length: 26
content-type: application/json
access-control-allow-origin: http://localhost:3000
access-control-expose-headers: x-app-version
vary: Origin
x-app-version: 1.0.190
date: Thu, 18 Jul 2019 17:43:20 GMT
{"error":"Internal error"}

{"level":"ERROR","time":"[2019-07-18][17:43:20]","target":"bigneon_api::big_neon_logger","message":"Cannot resend confirmation for unpaid order","api_version":"1.0.190","ip_address":"127.0.0.1:53071","method":"POST","uri":"/orders/ff99422b-f837-4ed6-9dde-74b5f8bc7a7a/resend_confirmation","user_id":"e5c294a2-47bc-4229-a525-131fb6536a6b"}
```

Buying 3 non discount, 2 with discount:
Screen Shot 2019-07-22 at 1 20 52 PM

Relevant part of order confirmation email:
Screen Shot 2019-07-22 at 1 23 23 PM

When communication blocked (showing variables):
```
{  
   "level":"INFO",
   "time":"[2019-07-22][17:21:08]",
   "target":"server",
   "message":"Blocked communication",
   "communication":{  
      "body":null,
      "categories":[  
         "purchase"
      ],
      "comm_type":"EmailTemplate",
      "destinations":{  
         "addresses":[  
            "nine+6@tari.com"
         ]
      },
      "extra_data":null,
      "source":{  
         "addresses":[  
            "noreply@bigneon.com"
         ]
      },
      "template_data":[  
         {  
            "item_breakdown":"<table style=\"width:100%\"><tbody><tr><th>Units</th><th>Description</th><th>Unit Price</th><th>Total</th></tr><tr><td align=\"center\">3</td><td>It's my party - Default_Pricing_1563808688</td><td align=\"right\">$30.00</td><td align=\"right\">$90.00</td></tr><tr><td align=\"center\">2</td><td>It's my party - General Admission_1563808688</td><td align=\"right\">$30.00</td><td align=\"right\">$60.00</td></tr><tr><td align=\"center\">2</td><td>Discount</td><td align=\"right\">($1.00)</td><td align=\"right\">($2.00)</td></tr></tbody></table>",
            "name":"Nine",
            "ticket_count":"5",
            "tickets_link":"http://localhost:3000/hub",
            "total_breakdown":"<table><tbody><tr><th>Fees Total</th><td>1.50</td></tr><tr><th>Order Total</th><td>149.50</td></tr></tbody></table>",
            "total_fees":"1.50",
            "total_initial_fees":"1.50",
            "total_price":"149.50",
            "total_refunded_fees":"0.00"
         }
      ],
      "template_id":"d-c23ba549dd0749bbb3b244b758c05dd7",
      "title":"BigNeon Purchase Completed"
   }
}
```

Order history page:
Screen Shot 2019-07-22 at 1 21 27 PM

Refunding one of each (discount and non discount):
Screen Shot 2019-07-22 at 1 21 48 PM
Screen Shot 2019-07-22 at 1 21 56 PM

Relevant part of order confirmation (note refunded items now):
Screen Shot 2019-07-22 at 1 24 20 PM

When communication blocked (showing variables):
```
{  
   "level":"INFO",
   "time":"[2019-07-22][17:23:35]",
   "target":"server",
   "message":"Blocked communication",
   "communication":{  
      "body":null,
      "categories":[  
         "purchase"
      ],
      "comm_type":"EmailTemplate",
      "destinations":{  
         "addresses":[  
            "nine+6@tari.com"
         ]
      },
      "extra_data":null,
      "source":{  
         "addresses":[  
            "noreply@bigneon.com"
         ]
      },
      "template_data":[  
         {  
            "item_breakdown":"<table style=\"width:100%\"><tbody><tr><th>Units</th><th>Description</th><th>Unit Price</th><th>Total</th></tr><tr><td align=\"center\">2</td><td>It's my party - General Admission_1563808688</td><td align=\"right\">$30.00</td><td align=\"right\">$60.00</td></tr><tr><td align=\"center\">2</td><td>Discount</td><td align=\"right\">($1.00)</td><td align=\"right\">($2.00)</td></tr><tr style=\"color: red\"><td align=\"center\">1</td><td>Refunded</td><td align=\"right\">($29.00)</td><td align=\"right\">($29.00)</td></tr><tr><td align=\"center\">3</td><td>It's my party - Default_Pricing_1563808688</td><td align=\"right\">$30.00</td><td align=\"right\">$90.00</td></tr><tr style=\"color: red\"><td align=\"center\">1</td><td>Refunded</td><td align=\"right\">($30.00)</td><td align=\"right\">($30.00)</td></tr></tbody></table>",
            "name":"Nine",
            "ticket_count":"3",
            "tickets_link":"http://localhost:3000/hub",
            "total_breakdown":"<table><tbody><tr><th>Fees Total</th><td>1.50</td></tr><tr style=\"color: red\"><th>Refunded</th><td>($0.20)</td></tr><tr><th>Order Total</th><td>90.30</td></tr></tbody></table>",
            "total_fees":"1.30",
            "total_initial_fees":"1.50",
            "total_price":"90.30",
            "total_refunded_fees":"0.20"
         }
      ],
      "template_id":"d-c23ba549dd0749bbb3b244b758c05dd7",
      "title":"BigNeon Purchase Completed"
   }
}
```

### Migrations
 * No Migrations

### Environment Variables
 * No change

### CODE REVIEW

**Comment by Timo614** [comment]
Set as DNM pending the questions from the issue which may modify some behavior / permissions.

**Comment by Timo614** [comment]
I've updated this to:
- Reflect the discounts in the order confirmation email (had a bug see above)
- Reflect refunds post refund as per the associated issue

Still DNM for now as I'm confirming if there are modifications to how this is displayed given I wasn't sure if I should make the discounts red as well / any other modifications we want to make there.

**Comment by Timo614** [comment]
Removed the DNM as I got a thumbs up for the new way of displaying the discounts in the associated issue. Should be good for review now.

File: api/src/communications/mailers/orders.rs [hunk 1]
```diff
@@ -1,11 +1,183 @@
...
+fn generate_item_row(
+    description: &str,
+    quantity: i64,
+    unit_price_in_cents: i64,
+    refund: bool,
+) -> String {
+    let mut item_row = "".to_string();
+
+    if refund {
+        item_row.push_str(r#"<tr style="color: red">"#);
```
Comment (mikethetike) [inline api/src/communications/mailers/orders.rs]: At some point in future we must investigate using handlebars templates in Sendgrid instead of raw html

**Pull request merged**